include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -71,15 +75,27 @@ class StatsdClient {
 
     //! The UDP sender to be used for actual sending
     std::unique_ptr<UDPSender> m_sender;
+
+    //! The random number generator for handling sampling
+    static thread_local std::mt19937 m_random_engine;
 };
 
+thread_local std::mt19937 StatsdClient::m_random_engine;
+
 inline StatsdClient::StatsdClient(const std::string& host,
                                   const uint16_t port,
                                   const std::string& prefix,
                                   const uint64_t batchsize) noexcept
     : m_prefix(prefix), m_sender(new UDPSender{host, port, batchsize}) {
+    // TODO: the final metric strings do not automatically place a '.' between the prefix and the key
+    //  This might be unexpected to end users. We should either document this or we could just add the
+    //  automatic '.' when the prefix is non empty
+    //if(!m_prefix.empty()) {
+    //    m_prefix.push_back('.');
+    //}
```

Comment:
it seemed a bit odd that we dont do this automatically. do you feel like we should or that we should document that its not automatic?